lua51Packages.nfd: fix module not loading

Petingoso opened the pull request:
As #295022 complains, it wasn't finding symbols properly. 
This PR fixes #295022 .

I traced it to the [migration](https://github.com/NixOS/nixpkgs/pull/288669) from buildLuarocksPackage.extraVariables to buildLuarocksPackage.luarocksConfig.variables where someone forgot to add .variables.

It now launches the dialogue properly.

## Description of changes
Added a missing property
<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

Petingoso committed:
lua51Packages.nfd: fix module not loading

As this (issue)[https://github.com/NixOS/nixpkgs/issues/295022] complains, it wasn't
finding symbols properly.

I traced it to the (migration)[https://github.com/NixOS/nixpkgs/pull/288669] from buildLuarocksPackage.extraVariables to
buildLuarocksPackage.luarocksConfig.variables where someone forgot to
add .variables.

It now launches the dialogue properly.

NotAShelf reviewed (approved):
LGTM!

teto commented:
could you run the tests to prevent further regressions please ? there seems to be busted tests. If that proves too hard, you could at least require the module. There are examples in lua/overrides.nix

Petingoso commented:
> could you run the tests to prevent further regressions please ? there seems to be busted tests. If that proves too hard, you could at least require the module. There are examples in lua/overrides.nix

I can't seem to find any appropriate tests under nixos tests, under the pkgs.lua51packages.nfd.passthru.tests nor in pkgs/tests.

I'm not quite sure about what module you're talking about overriding.

I apologize for my lack of experience with all of these frameworks

teto commented:
sry if that was not clear, my suggestion was to add tests to this nix package because there are none while the source itself seems to have busted tests at least https://github.com/Alloyed/nativefiledialog/tree/master/lua/spec .
You can see how it's done at
https://github.com/NixOS/nixpkgs/blob/73cb87fe1534b9e3a37a6c3e04e7c1001f464044/pkgs/development/lua-modules/overrides.nix#L66
(hopefully it will get easier).
No commit in seven years though.

Petingoso commented:
> sry if that was not clear, my suggestion was to add tests to this nix package because there are none while the source itself seems to have busted tests at least https://github.com/Alloyed/nativefiledialog/tree/master/lua/spec . You can see how it's done at
> 
> https://github.com/NixOS/nixpkgs/blob/73cb87fe1534b9e3a37a6c3e04e7c1001f464044/pkgs/development/lua-modules/overrides.nix#L66
> 
> (hopefully it will get easier).
> No commit in seven years though.

It actually pulls from a more modern fork, https://github.com/Vexatos/nativefiledialog/tree/master/test.
Issue is I'm trying to go through the different phases via a nixshell but being everything wrapped in buildLuaRocksPackage I'm not understanding how to go about it as at the time of CheckPhase i don't seem to have anything built.

Right now when i trigger a CheckPhase it seems luarocks complains `Argument missing: please specify a rockspec to use on current directory.`

Being the snippet reproducible with just `doCheck = true;` inside buildLuarocksPackage

Petingoso committed:
lua51Packages.nfd: feat added postInstallCheck for test

lua51Packages.nfd: cleanup duplicate line

lua51Packages.nfd: cleanup useless undefined hooks

lua51Packages.nfd: squashed commits

lua51Packages.nfd: change to generic buster

Petingoso commented:
After wrestling with this, I think i have tests working

UlyssesZh commented:
Please add "fix #295022" in the PR comment so that GitHub will automatically close that issue when this PR gets merged.

UlyssesZh commented:
For small fixing commits like cba74580b, please squash into previous commits. See [commit conventions](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#commit-conventions).

teto commented on pkgs/development/lua-modules/nfd/default.nix:
```diff
@@ -22,15 +21,22 @@ buildLuarocksPackage {
   '';
 
+  doInstallCheck = true;
+  installCheckInputs = [ lua51Packages.busted ];
```
use `lua.pkgs.busted` instead since this will mix a 5.1 version when installing lua_52.pkkgs.nfd

teto commented:
thank you
